### PR TITLE
Add shows detail root view controller

### PR DESCRIPTION
**Pull request by javamonn**
Inspired by #1198 and [documentation](https://github.com/artsy/emission/blob/master/docs/adding_new_components.md), adds a new ViewController and react component scaffolds for the shows detail page. Implements [LD-11](https://artsyproduct.atlassian.net/browse/LD-11). Implemented while pairing with @l2succes and @ashleyjelks.

#skip_new_tests (We'll add tests as we build those components)

### CODE REVIEW

File: src/lib/Scenes/ReferenceShow/index.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        keyExtractor={(item, index) => item.type + String(index)}
+      />
+    )
+  }
+}
+
+export default createFragmentContainer(
+  ReferenceShow,
+  graphql`
+    fragment ReferenceShow_show on Show {
```
Comment (javamonn) [inline src/lib/Scenes/ReferenceShow/index.tsx]: There's some churn between `ReferenceShow` and `Show` for this container. We'd like to use `Show`, but have to refactor the [existing conflicting definition](https://github.com/javamonn/emission/blob/bdd134aa35b520bbacba0d642400043992c1bb3e/src/lib/Components/Artist/Shows/Show.tsx#L46). I got partway through doing this in this PR but reverted after running into some unrelated issues, thinking I'll follow open a follow up PR to make the naming more consistent.

Comment (javamonn) [reply src/lib/Scenes/ReferenceShow/index.tsx]: I discussed this with @alloy a bit and he told me about coming to the decision of applying better namespacing to fragments. I'll rename this to something like `LocalDiscoveryShow`, and we can ticket the follow up work to namespace other fragments, e.g.`Show_show` should become something like `ArtistsShow_show`

Comment (javamonn) [reply src/lib/Scenes/ReferenceShow/index.tsx]: Discussed this a bit more with @l2succes and ended up waffling - the local discovery show is intended to be the canonical showI think, so it seems to make sense to namespace the Artist's `show`. I'm open to discussion though.

**Comment by l2succes** [comment]
merge on green

**Comment by alloy** [comment]
Gotcha, makes sense 👍

**Pull request merged**